src/vision/status_quo/frederik_makes_their_first_steps_into_async.md

```diff
@@ -0,0 +1,64 @@
+
+### Woes of a newcomer to async
+Being reasonably confident in their choice of ecosystem, Frederik starts building their distributed system.
+After a while, they want to introduce another networking library of which the api isn't async. Luckily Frederik picked up on that blocking was not allowed in async (or at least not in any of the currently existing executors), through reading some blog posts about async. More reddit discussions point them towards spawn\_blocking in Tokio, and even rayon. But they're none the wiser about how to apply these paradigms in a neat manner.
+
+Previously the design patterns learned in other languages, combined with the patterns thaught in the book, were usually sufficient to come to reasonably neat designs.
+But neither their previous experience, nor the async book nor the Tokio tutorial were of much use when trying to neatly incorporate blocking code into their previously fully async project.
+
+### Confused ever after
+To this day the lack of a blessed approach leaves Frederik unsure about the choices they've made so far and misconceptions they might still have, evermore wondering if the original tweets they read about most how most people should just stick to threads were right all along.
+
+🤔 Frequently Asked Questions
+
+* **What are the morals of the story?**
+    * When entering Rust's async world without previous async experience, and no benchmarks for what good async design patters look like, getting started with async can be a bit overwhelming.
+    * Other languages which only have a single ecosystem seem to have a much better story for beginners since there's no fear of lock in, or ecosystem fomo about making the wrong choices early on.
+* **What are the sources for their story?**
+    * Various blog posts of withoutboats
+    * A blog post which spurred a lot of discussion about blocking in async: https://async.rs/blog/stop-worrying-about-blocking-the-new-async-std-runtime/
+    * A nice blog post about blocking in Tokio, which still doesn't have any nice design patterns: https://ryhl.io/blog/async-what-is-blocking/
+    * An example of design patterns being discussed for sync Rust in the book: https://doc.rust-lang.org/book/ch17-03-oo-design-patterns.html#trade-offs-of-the-state-pattern
+    * Perhaps I should've read a bit more of Niko's blogs and his async interviews.
+* **Why did you choose *NAME* to tell their story?**
+    * Don't know yet.
+* **How would their story have played out differently for the other characters?**
+    * Characters with previous async experience would probably have had a much better experience getting started with async in Rust, so Barbara probably wouldn't experience this problem.
```

Comment:
Barbara's character profile doesn't mention any experience with any other language, so I think Barbara could fit nicely with this story. In fact, part of Frederik "paranoia" was knowing enough about the ecosystem already to know that lock-in was a potential issue. A complete newcomer might not know enough for this to even be an issue.